python_modules/dagster/dagster/core/storage/schedules/sqlite/sqlite_schedule_storage.py

```diff
@@ -43,22 +43,29 @@ def config_type(cls):
     def from_config_value(inst_data, config_value):
         return SqliteScheduleStorage.from_local(inst_data=inst_data, **config_value)
 
-    @staticmethod
-    def from_local(base_dir, inst_data=None):
+    @classmethod
+    def from_local(cls, base_dir, inst_data=None):
         check.str_param(base_dir, "base_dir")
         mkdir_p(base_dir)
         conn_string = create_db_conn_string(base_dir, "schedules")
         engine = create_engine(conn_string, poolclass=NullPool)
         alembic_config = get_alembic_config(__file__)
 
+        should_migrate_data = False
         with engine.connect() as connection:
             db_revision, head_revision = check_alembic_revision(alembic_config, connection)
             if not (db_revision and head_revision):
                 ScheduleStorageSqlMetadata.create_all(engine)
                 engine.execute("PRAGMA journal_mode=WAL;")
                 stamp_alembic_rev(alembic_config, connection)
+                should_migrate_data = True
+
+        schedule_storage = cls(conn_string, inst_data)
+        if should_migrate_data:
+            schedule_storage.migrate()
+            schedule_storage.optimize()
```

Comment:
is there any risk of things failing partway through and leaving the schema created but the data migrations not run?